File: src/main/kotlin/ps/api/user/atomizer/dto/CreateDto.kt
package ps.api.user.atomizer.dto

import kotlinx.serialization.Serializable
import leight.client.sdk.annotation.TypeBool
import leight.client.sdk.annotation.TypeNumber
import leight.client.sdk.annotation.TypeString
import leight.sdk.annotation.Module

@Serializable
@Module("user/atomizer")
data class CreateDto(
	@TypeString
	val name: String,
	@TypeString
	val code: String,
	@TypeString
	val vendorId: String,
	@TypeNumber(nullable = true)
	val coils: Int? = null,
	@TypeNumber(nullable = true)
	val maxCoilSize: Int? = null,
	@TypeNumber(nullable = true)
	val maxWraps: Int? = null,
	@TypeNumber(nullable = true)
	val capacity: Float? = null,
	@TypeBool
	val squonk: Boolean,
	@TypeNumber(nullable = true)
	val base: Int? = null,
)

File: src/PuffSmith/Base/Dto/Create/CreateDto.php
<?php
declare(strict_types=1);

namespace PuffSmith\Base\Dto\Create;

use Edde\Dto\AbstractDto;

class CreateDto extends AbstractDto {
	/**
	 * @var string
	 */
	public string $name;
	/**
	 * @var int
	 */
	public int $pg;
	/**
	 * @var int
	 */
	public int $vg;
	/**
	 * @var string
	 */
	public string $vendorId;
}
